core/integration/odc/fairmq/fairmq.go

```diff
@@ -0,0 +1,80 @@
+/*
+ * === This file is part of ALICE O² ===
+ *
+ * Copyright 2024 CERN and copyright holders of ALICE O².
+ * Author: Teo Mrnjavac <teo.mrnjavac@cern.ch>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * In applying this license CERN does not waive the privileges and
+ * immunities granted to it by virtue of its status as an
+ * Intergovernmental Organization or submit itself to any jurisdiction.
+ */
+
+package fairmq
+
+var (
+	fairMqStateMap map[string]string
+)
+
+func init() {
+	// see https://github.com/FairRootGroup/FairMQ/blob/master/docs/images/device_states.svg
+	fairMqStateMap = map[string]string{
+		"IDLE":    "STANDBY",
+		"READY":   "CONFIGURED",
+		"RUNNING": "RUNNING",
+		"ERROR":   "ERROR",
+		"EXITING": "DONE",
+
+		"INITIALIZING DEVICE":      "STANDBY",
+		"INITIALIZED":              "STANDBY",
+		"BINDING":                  "STANDBY",
+		"BOUND":                    "STANDBY",
+		"CONNECTING":               "STANDBY",
+		"DEVICE READY CONFIGURING": "STANDBY", // needs special case
+		"INITIALIZING TASK":        "STANDBY",
+
+		"RESETTING TASK":         "CONFIGURED",
+		"DEVICE READY RESETTING": "CONFIGURED", // needs special case
+		"RESETTING DEVICE":       "CONFIGURED",
```

Comment:
Fair enough, and I can implement this, though I guess in that case instead of looking at the previous ODC state we should account for which ODC state changes are invariant with respect to the ECS state.

AliECS cannot affect ODC states with the granularity you describe, so for this to happen ODC would've had to reach `INITIALIZED` and then go back instead of forward on its own accord contrary to AliECS input. In any case we end up back in `IDLE` or possibly `ERROR` so not much changes for AliECS in between.

For this matter, then even the case of `IDLE`→`INITIALIZED`→`BOUND`→`DEVICE READY`→`RESETTING DEVICE` (similarly not requestable by AliECS) should also resolve to `STANDBY`, and in order to implement this we'd need to look back 2 states instead of 1 as we do now (or forget about ODC state lookback and take ECS state into account instead). When the interface was negotiated with ODC we explicitly demanded a single call to go from `IDLE` to `READY`, and for ODC to handle any mess that might happen in between.